### PR TITLE
[Driver] Hook up Haiku ARM support

### PR DESCRIPTION
**Pull request by brad0**

### CODE REVIEW

**Comment by brad0** [comment]
@MaskRay

File: clang/test/Driver/arm-abi.c [hunk 1]
```diff
@@ -33,6 +33,8 @@
 // RUN:   | FileCheck -check-prefix=CHECK-AAPCS-LINUX %s
 // RUN: %clang -target arm--openbsd- %s -### -o %t.o 2>&1 \
 // RUN:   | FileCheck -check-prefix=CHECK-AAPCS-LINUX %s
+// RUN: %clang -target arm--haiku- %s -### -o %t.o 2>&1 \
```
Comment (MaskRay) [inline clang/test/Driver/arm-abi.c:36]: This check can be moved to `haiku.c`. Having most Haiku-related checks in one place seems cleaner.

**Review by MaskRay — APPROVED** [review]
Hope that a Haiku developer can confirm this is what they want:)

**Comment by MaskRay** [comment]
> @MaskRay

Will be out of town for ~2 weeks and will be slow to respond

**Comment by brad0** [comment]
@davidkaroly

**Comment by brad0** [comment]
@nielx

**Pull request merged**